index.rst

```diff
@@ -4,9 +4,184 @@ File Formats and Layouts for Cell-based Coadds
+
+Exploded Images
+---------------
+
+This approach is similar to the data cube layout, with one HDU for each image plane, but instead of using additional dimensions to represent the grid, we just stitch all cells into a single larger 2-d image.
+This doesn't put the cells onto a consistent meaningful coordinate system, however, because we'd be stitching the outer regions together, not the inner regions, and that means all of the logical pixels in the overlap regions appear more than once (albeit with subtly different PSFs, noise, etc, due to different input epochs, in most cases).
+
+That makes the full image *somewhat* interpretable by humans, though far from ideal - it's a bit similar to the common approach of displaying "untrimmed" raw images with the overscan regions of amplifiers in between the data sections.
+This is a slight advantage over the data cube layout, but it seems to be the only one: it suffers from the same incompatibility with FITS WCS, and is similarly neutral for the compressed subimage reads problem.
+
+Stitched Images
+---------------
+
+We expect most accesses to our coadd files to be uninterested in the redundant overlap pixel values - instead, most science users will be interested in stitching together the inner cells to form a mostly-seamless patch-level image.
+This suggests that we might want to store that stitched inner-cell image for each plane directly as a single HDU, and shunt the overlap pixel values to a different HDU.
+
+For the inner-cell image, this is ideal: FITS WCS can be used exactly the way it was intended, and third-party FITS viewers will be completely usable without any extra effort.
+
+For the overlap regions, we'd end up with a repeat of our original problem, but with lower stakes: for each original cell, we'd have 4 overlap-region images (top, bottom, left, right) that need to be packed into a binary table, data cube, or stitched image of their own (with this stitching being analogous to the exploded coadd case, since there'd be no meaningful overall coordinate system).
+
+.. figure:: /_static/cell-stitching.png
+   :name: cell-stitching
+   :target: ../_images/cell-stitching.png
+
+   Left: an "exploded" image with inner cell regions (squares) and a subdivision of the overlap regions (T, B, L, R) labeled.
+
+   Right: a stitched inner-cell image and packings of the overlap regions.
+   Each color represents a different FITS HDU; the green inner region would be a single 2-d image, while the blue and purple overlap regions could 2-d images or 3- or 4-d data cubes.
+
+Assuming we don't care about FITS WCS support for the overlap regions, the main problem with this approach is complexity in two places:
+
+- When writing, we'd need to quantize the outer cell image first, and then slice the image into its inner-cell and overlap-region sections, and only then compress the quantized values.
```

Comment:
Can you clarify (either as a comment here, or in the document) if you mean to compress not just the inner cell, but _also_ the overlap regions? And if so, can just the inner cell be decompressed and read without ever having to access the overlap regions?

Reply:
So to be very clear, you need to quantize separately per cell. Compression together is fine though. This is what is in your text above but I wanted to state it very plainly to ensure we're on the same page.